cmd/svcat/broker/register_cmd.go

```diff
@@ -18,57 +18,122 @@ package broker
-		"The broker URL (Required)")
-	cmd.MarkFlagRequired("url")
+	cmd.Flags().StringVar(&registerCmd.BasicSecret, "basic-secret", "",
+		"A secret containing basic auth (username/password) information to connect to the broker")
+	cmd.Flags().StringVar(&registerCmd.BearerSecret, "bearer-secret", "",
+		"A secret containing a bearer token to connect to the broker")
+	cmd.Flags().StringVar(&registerCmd.CAFile, "ca", "",
+		"A file containing the CA certificate to connect to the broker")
+	cmd.Flags().StringSliceVar(&registerCmd.ClassRestrictions, "class-restrictions", []string{},
+		"A whitelist of Classes to allow from the broker")
+	cmd.Flags().StringSliceVar(&registerCmd.PlanRestrictions, "plan-restrictions", []string{},
+		"A whitelist of Plans to allow from the broker")
+	cmd.Flags().StringVar(&registerCmd.RelistBehavior, "relist-behavior", "",
+		"Behavior for relisting the broker's catalogue. Manual/Duration")
+	cmd.Flags().DurationVar(&registerCmd.RelistDuration, "relist-duration", 0*time.Second,
+		"Interval to refetch broker catalog when relist-behavior is set to duration, specified in human readable format: 30s, 1m, 1h")
+	cmd.Flags().BoolVar(&registerCmd.SkipTLS, "skip-tls", false,
+		"Disables TLS certificate verification when communicating with this broker. This is strongly discouraged. You should use --ca instead.")
+	registerCmd.AddWaitFlags(cmd)
+
 	return cmd
 }
 
 // Validate checks that the required arguements have been provided
 func (c *RegisterCmd) Validate(args []string) error {
-	if len(args) == 0 {
-		return fmt.Errorf("a broker name is required")
+	if len(args) < 2 {
+		return fmt.Errorf("a broker name and url is required")
 	}
 	c.BrokerName = args[0]
+	c.URL = args[1]
 
+	if c.BasicSecret != "" && c.BearerSecret != "" {
+		return fmt.Errorf("cannot use both basic auth and bearer auth")
+	}
+
+	if c.CAFile != "" {
+		_, err := os.Stat(c.CAFile)
+		if err != nil {
+			return fmt.Errorf("error finding CA file: %v", err.Error())
+		}
+	}
+	if c.RelistBehavior != "" {
+		c.RelistBehavior = strings.ToLower(c.RelistBehavior)
+		if c.RelistBehavior != "duration" && c.RelistBehavior != "manual" {
+			return fmt.Errorf("relist-behavior must be duration/manual")
+		}
+	}
+	if c.RelistDuration != 0*time.Second && c.RelistBehavior != "duration" {
+		return fmt.Errorf("relist-behavior must be duration to set relist-duration")
+	}
 	return nil
 }
 
-// Run runs the command
+// Run calls out to the pkg lib to create the broker and displays the output
 func (c *RegisterCmd) Run() error {
-	return c.Register()
-}
 
-// Register calls out to the pkg lib to create the broker and displays the output
-func (c *RegisterCmd) Register() error {
-	broker, err := c.Context.App.Register(c.BrokerName, c.URL)
+	opts := &servicecatalog.RegisterOptions{
+		BasicSecret:       c.BasicSecret,
+		BearerSecret:      c.BearerSecret,
+		CAFile:            c.CAFile,
+		ClassRestrictions: c.ClassRestrictions,
+		Namespace:         c.Context.App.CurrentNamespace,
```

Comment:
Do not directly use the current namespace, because that doesn't properly take into account all of the necessary logic. The `Namespaced` struct handles defaulting this appropriately, and the final namespace can be found in `c.Namespace`.

Reply:
how does that interact with throwing the global kubectl namespace flag when running in plugin mode?